src/pymor/algorithms/greedy.py

```diff
@@ -81,73 +87,104 @@ def greedy(discretization, reductor, samples, initial_basis=None, use_estimator=
-    max_errs = []
-    max_err_mus = []
-    hierarchic = False
-
-    rd, rc, reduction_data = None, None, None
-    while True:
-        logger.info('Reducing ...')
-        rd, rc, reduction_data = reductor(discretization, basis) if not hierarchic \
-            else reductor(discretization, basis, extends=(rd, rc, reduction_data))
-
-        if len(samples) == 0:
-            logger.info('There is nothing else to do for empty samples.')
-            return {'basis': basis, 'reduced_discretization': rd, 'reconstructor': rc,
-                    'max_errs': [], 'max_err_mus': [], 'extensions': 0,
-                    'time': time.time() - tic, 'reduction_data': reduction_data}
-
-        logger.info('Estimating errors ...')
-        if use_estimator:
-            errors = [rd.estimate(rd.solve(mu), mu) for mu in samples]
-        elif error_norm is not None:
-            errors = [error_norm(discretization.solve(mu) - rc.reconstruct(rd.solve(mu))) for mu in samples]
-        else:
-            errors = [(discretization.solve(mu) - rc.reconstruct(rd.solve(mu))).l2_norm() for mu in samples]
-
-        # most error_norms will return an array of length 1 instead of a number, so we extract the numbers
-        # if necessary
-        errors = map(lambda x: x[0] if hasattr(x, '__len__') else x, errors)
-
-        max_err, max_err_mu = max(((err, mu) for err, mu in izip(errors, samples)), key=lambda t: t[0])
-        max_errs.append(max_err)
-        max_err_mus.append(max_err_mu)
-        logger.info('Maximum error after {} extensions: {} (mu = {})'.format(extensions, max_err, max_err_mu))
-
-        if target_error is not None and max_err <= target_error:
-            logger.info('Reached maximal error on snapshots of {} <= {}'.format(max_err, target_error))
-            break
-
-        logger.info('Extending with snapshot for mu = {}'.format(max_err_mu))
-        U = discretization.solve(max_err_mu)
-        try:
-            basis, extension_data = extension_algorithm(basis, U)
-        except ExtensionError:
-            logger.info('Extension failed. Stopping now.')
-            break
-        extensions += 1
-        if 'hierarchic' not in extension_data:
-            logger.warn('Extension algorithm does not report if extension was hierarchic. Assuming it was\'nt ..')
-            hierarchic = False
-        else:
-            hierarchic = extension_data['hierarchic']
-
-        logger.info('')
-
-        if max_extensions is not None and extensions >= max_extensions:
-            logger.info('Maximum number of {} extensions reached.'.format(max_extensions))
-            logger.info('Reducing once more ...')
+    sample_count = len(samples)
+    logger.info('Started greedy search on {} samples'.format(sample_count))
+    if pool is None or pool is dummy_pool:
+        pool = dummy_pool
+    else:
+        logger.info('Using pool of {} workers for parallel greedy search'.format(len(pool)))
+
+    with RemoteObjectManager() as rom:
```

Comment:
`rom` may be a misleading name in the context of model reduction (_reduced order model_), but it is probably unproblematic in this local context...